### ISSUE TITLE
speedtest is not working anymore in latest version

### ISSUE THREAD

**Issue by crazychatting** [roest01/docker-speedtest-analyser#74]
hi,

i noticed that the new "latest" tagged image from 5 days ago throws some errors from ngingx and the results are not working anymore.

### Log output:
> Starting run.sh
> Starting Cronjob
> Starting nginx
> nginx: [alert] detected a LuaJIT version which is not OpenResty's; many optimizations will be disabled and performance will be compromised (see https://github.com/openresty/luajit2 for OpenResty's LuaJIT or, even better, consider using the OpenResty releases from https://openresty.org/en/download.html)
> nginx: [error] lua_load_resty_core failed to load the resty.core module from https://github.com/openresty/lua-resty-core; ensure you are using an OpenResty release from https://openresty.org/en/download.html (rc: 2, reason: module 'resty.core' not found:
> 	no field package.preload['resty.core']
> 	no file './resty/core.lua'
> 	no file '/usr/share/luajit-2.1.0-beta3/resty/core.lua'
> 	no file '/usr/local/share/lua/5.1/resty/core.lua'
> 	no file '/usr/local/share/lua/5.1/resty/core/init.lua'
> 	no file '/usr/share/lua/5.1/resty/core.lua'
> 	no file '/usr/share/lua/5.1/resty/core/init.lua'
> 	no file '/usr/share/lua/common/resty/core.lua'
> 	no file '/usr/share/lua/common/resty/core/init.lua'
> 	no file './resty/core.so'
> 	no file '/usr/local/lib/lua/5.1/resty/core.so'
> 	no file '/usr/lib/lua/5.1/resty/core.so'
> 	no file '/usr/local/lib/lua/5.1/loadall.so'
> 	no file './resty.so'
> 	no file '/usr/local/lib/lua/5.1/resty.so'
> 	no file '/usr/lib/lua/5.1/resty.so'
> 	no file '/usr/local/lib/lua/5.1/loadall.so')
> --- running speedtest ---
> --- running speedtest ---

also the graph seems to be broken now, shows always "invalid date" and has no result points in it. the console in the browser says
> GET http://[ip]/data/appConfig.js net::ERR_ABORTED 404 (Not Found)

what can i do?

**Comment by crazychatting** [comment]
I also tried image tag v1.2 and develop. the error from the log output is gone there but still shows no results? tried even to clear all the saved results to prevent incompatibility...
also cleared browser cache of course ;)

**Comment by meiro-zz** [comment]
Same here!

**Comment by jaxjexjox** [comment]
I have gone into the container and confirmed it has rights, to write to results.csv however it's failing to do so.
I'm unsure why, I suspect it was my fault but seemingly not.
I deleted my history, I edited the file, I changed permissions, I confirmed file writes possible while  exec -i into the container.

Is there a similar product out there? I like this one quite a bit, it's nice and simple - but it has broken a few times now

**Comment by GitBeto21** [comment]
> Is there a similar product out there? I like this one quite a bit, it's nice and simple - but it has broken a few times now

be careful what you say.  Roest does not accept negative feedback. maybe you (and I) will be banned out! 😎

**Comment by jaxjexjox** [comment]
I'm not giving negative feedback, I'm just enquiring if there's other options.   If I recall this was the best one which 'rolled its own database' (even if it's a CSV)      
Some of the other ones are unnecessarily complicated.

I helped report a bug previously, no one is getting banned.

**Comment by roest01** [comment]
i am on this bug. unfortunately i have not a clear idea why this is happening. In my opinion this tool should work with */0 and */30 as well. I am refactoring and updating deps right now on development branch. 

If anyone has a clear idea please help me out. Meanwhile negative feedback is not a problem ;) this project was a glance before getting a hard rock.

**Comment by jaxjexjox** [comment]
I am not seeing content, written to the csv file, however the 'invalid dates' thing someone mentioned was interesting.

**Comment by Bananas-Are-Yellow** [comment]
I tried the docker image for the _develop_ branch, but this doesn't work either.

I cleared my browser cache and now I get an error saying `Chart.js` is not found. It is this line that fails:
```
<script src="node_modules/chart.js/dist/Chart.js"></script>
```

**Comment by tuckshoprn** [comment]
The file name seems to have a capital C rather than a small one, I’d try that and see what happens!

**Comment by Bananas-Are-Yellow** [comment]
> The file name seems to have a capital C rather than a small one, I’d try that and see what happens!
> […](#)
> Sent from my iPhone
> On Apr 26, 2021, at 09:15, David Taylor ***@***.***> wrote: ﻿ I tried the docker image for the develop branch, but this doesn't work either. I cleared my browser cache and now I get an error saying Chart.js is not found. It is this line that fails: <script src="node_modules/chart.js/dist/Chart.js"></script> — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

You are right, that's the problem. With a lowercase `c` the resource is found:
```
/node_modules/chart.js/dist/chart.js
```

**Comment by jaxjexjox** [comment]
Well done all, thank you. Looks like a simple mistake!

**Comment by github-iw** [comment]
Seems linked to a change in speedtest: https://www.raspberrypi.org/forums/viewtopic.php?t=309024

**Comment by crazychatting** [comment]
@github-iw I don´t think that this is related to the other change... so far the latest commit in this repository here was on April 12 (cf6bd4f919182b0de7ed84f9c0f569c960b0bc4c), since when the docker container is not working anymore.
maybe it is related to the update of the yarn dependency?

the commit before this one was in september 2020, and I can´t really see a relation to other projects like the one you mentioned.

**Comment by github-iw** [comment]
I’m currently running 2 versions, the v1.2 to show the graphs online (v1.2 does not run the speedtest anymore on my Synology but still shows the correct graphs), the latest to schedule the hourly test (this version does run the Cronkite speedtest but does not show the graph (invalid dates)). Both directed to the same network share for the results file.

**Comment by jaxjexjox** [comment]
Is this problem actually fixed? My copy continues to not work, despite the fact, I run this command routinely every couple of days for the past 6 months.

sudo docker pull roest/docker-speedtest-analyser  && sudo docker stop speedtestanal && sudo docker rm speedtestanal && sudo docker run -d    --name speedtestanal    -p 8989:80    -v /etc/localtime:/etc/localtime    -v /dockerconfigs/speedtestanal:/var/www/html/data/    --restart unless-stopped    -e CRONJOB_ITERATION=30    roest/docker-speedtest-analyser 


(ie: I update it manually, with regularity)

**Comment by MrCaringi** [comment]
Same here... lastest version doenst work:
`(rc: 2, reason: module 'resty.core' not found:`

**Comment by pfaeffli** [comment]
I do not think the problem lays within this repository itself. As far as I understand had nginx some bug fix which now causes this error.

Here some references:

* https://github.com/linuxserver/docker-swag/issues/21 refers to `nginx: [alert] detected a LuaJIT`
* https://github.com/openresty/lua-nginx-module/issues/1533 identical error with a suggested solution. I tried the solution mentioned there but failed so far.

I have no experience with nginx. I could install in the docker image openresty instead of LuaJIT. But I do not understand the implication of that for this repository. Would that solve this error or cause more trouble?

**Comment by jaxjexjox** [comment]
For anyone curious I have added this to my docker host.
https://hub.docker.com/r/henrywhitaker3/speedtest-tracker

It's not as good as this one as I prefer the layout of analyser, however analyser has been broken for at least 3 months for me now I think, so I'm just leaving it running (broken) and will switch back to it, if it's ever fixed.

**Comment by MrCaringi** [comment]
> For anyone curious I have added this to my docker host.
> https://hub.docker.com/r/henrywhitaker3/speedtest-tracker
> 
> It's not as good as this one as I prefer the layout of analyser, however analyser has been broken for at least 3 months for me now I think, so I'm just leaving it running (broken) and will switch back to it, if it's ever fixed.

thank you very much! I'll try it!

**Comment by jaxjexjox** [comment]
I have now also added this one too.
https://github.com/ansemjo/speedtest-plotter
